### PR TITLE
Adds jupyter-lsp, python-lsp-server, and notebook to 1.0.0 beta

### PR DESCRIPTION
**Pull request by JasonWeill**
*Description of changes:* Adds [jupyter-lsp](https://github.com/jupyter-lsp/jupyterlab-lsp), [python-lsp-server](https://github.com/python-lsp/python-lsp-server) and [notebook](https://github.com/jupyter/notebook/) packages to the 1.0.0-beta distribution.

This change supersedes #68, which had targeted `main`.

Addition to `cpu.env.in` and `gpu.env.in`:

```
conda-forge::jupyter-lsp[version='>=2.2.0,']
conda-forge::python-lsp-server[version='>=1.8.2,']
conda-forge::notebook[version='>=7.0.0,']
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by JasonWeill** [comment]
New tests have passed:

```
platform linux -- Python 3.11.6, pytest-7.4.2, pluggy-1.3.0 -- /home/jweill/miniconda3/envs/sagemaker-distribution/bin/python3.11
cachedir: .pytest_cache
rootdir: /local/home/jweill/git/sagemaker-distribution
configfile: pytest.ini
plugins: xdist-3.3.1, mock-3.11.1
collected 25 items / 23 deselected / 2 selected                                                                                                       

test/test_dockerfile_based_harness.py::test_dockerfiles_for_cpu[jupyter-lsp.test.Dockerfile-required_packages0] PASSED                          [ 50%]
test/test_dockerfile_based_harness.py::test_dockerfiles_for_cpu[notebook.test.Dockerfile-required_packages1] PASSED                             [100%]

================================================================== warnings summary ===================================================================
src/package_staleness.py:1
  /local/home/jweill/git/sagemaker-distribution/src/package_staleness.py:1: PendingDeprecationWarning: conda.cli.python_api is pending deprecation and will be removed in 24.9. Use `conda.testing.conda_cli` instead.
    import conda.cli.python_api

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

**Comment by JasonWeill** [comment]
@balajisankar15 I'd appreciate if you could take a look at this. Thanks!

File: test/test_artifacts/v1/jupyter-lsp.test.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ARG COSMOS_IMAGE
+FROM $COSMOS_IMAGE
+
+ARG MAMBA_DOCKERFILE_ACTIVATE=1
+
+CMD ["python", "-c", "import jupyter_lsp"]
```
Comment (just4brown) [inline test/test_artifacts/v1/jupyter-lsp.test.Dockerfile:6]: import validation is a good start, but typically we're trying to go deeper with validation for these test cases if possible. If you take a look at other packages' respective docker test files you'll find different approaches such as cloning OSS git repositories where unit test suites can be executed, or in other cases executing a set of jupyter notebook files

**Review by just4brown — CHANGES_REQUESTED** [review]

File: test/test_artifacts/v1/jupyter-lsp.test.Dockerfile (hunk 1 above)
Comment (JasonWeill) [reply test/test_artifacts/v1/jupyter-lsp.test.Dockerfile:6]: I spoke with @just4brown about this. It's impractical to test these particular packages in other ways, but for packages like altair (#84) we can execute a sample notebook, for example.

**Review by just4brown — APPROVED** [review]

**Pull request merged**